### ISSUE TITLE
Module export not read (AMD: define is not a function)

### ISSUE THREAD

**Issue by OliverJAsh** [circlingthesun/systemjs-script-plugin#1]
When I use this plugin to load an AMD module, I get the following error:

![image](https://cloud.githubusercontent.com/assets/921609/7275443/e359b42a-e8fb-11e4-8b36-668a1c025d75.png)

Likewise, if I use this script to load a plugin that exports a global, the global is not returned.

I am failing to see why this doesn't work yet the CSP version of SystemJS does, when the code is almost identical: https://github.com/systemjs/systemjs/blob/master/lib/extension-scriptLoader.js

/cc @guybedford @phamann

**Comment by circlingthesun** [comment]
This plugin adds the script tag with the module url to the page and removes it once loaded. If define is not in the global scope, you will get the error above.  As far as I know the CSP version assumes that the module you are loading was created by systemjs-builder, which probably then uses systemjs define calls instead of a global define.

If you run the lines below after loading systemjs it should work:

``` js
window.define = System.amdDefine;
window.require = System.amdRequire;
```

**Comment by OliverJAsh** [comment]
@guybedford Would it be possible to write a plugin that downloads using the
script tag _and_ supports AMD/CJS/globals? I understand there's no way ES6
modules can work, because we can't do any transpiling when downloading via
script.

**Comment by guybedford** [comment]
@OliverJAsh the next version of SystemJS will always use Script tags to load AMD and System.register, but globals, CJS and ES6 do need to always be loaded via XHR.

**Comment by OliverJAsh** [comment]
@phamann So for now, we could just run off the master version of SystemJS
which has this fix already…

**Comment by guybedford** [comment]
@OliverJAsh @phamann I would do my best to assist if running on master, but it's not advisable as it is not yet stable, and there will be bugs. The alternative I mentioned was to back port just this one feature into your build of SystemJS. If you would like me to assist with this, I'm happy to discuss further.

**Comment by circlingthesun** [comment]
It would be great if globals could also be loaded remotely via script tag without a plugin, or manually inserting script tags. Facebook, Google, and many others have libraries that one is expected to load remotely, use globals and cannot be loaded via XHR because of CORS.

**Comment by guybedford** [comment]
Yeah I guess this would be useful in its small subset - I've posted https://github.com/systemjs/systemjs/issues/420.

**Comment by OliverJAsh** [comment]
Surely globals can already be downloaded using this plugin/the CSP version
of System? The thing we need to solve is actually returning the global as
an export.

**Comment by guybedford** [comment]
Yes exactly, this is purely about maintaining compatibility with the expectations of the other features, by strictly defining the cases where it applies.
